src/Portfolio.sol

```diff
@@ -46,12 +54,100 @@ contract Portfolio is Ownable {
     }
 
     /**
-     * @notice Adds a new asset to the portfolio
+     * @notice Deposit funds for a portfolio share. This rebalances the amount of shares each owner has.
+     * @param tokenId The ID of the token to deposit funds for.
+     */
+    function deposit(uint256 tokenId) public payable {
+        require(ownerOf(tokenId) == msg.sender, "You are not the owner of this share");
+        uint256 previousValue = getPortfolioValue();
+
+        // register the deposited ETH in the balances
+        uint256 depositedAmount = msg.value;
+        assets["ETH"].balance += depositedAmount;
```

Comment:
Avoid using hardcoded values, this should allow any token

Reply:
I would prefer that in an initial version, only ETH can be deposited and withdrawn from the contract.

Reply:
Sounds reasonable, but let's keep the "ETH" string as a constant and no hardcoded.